File: blueluak-jse/src/test/java/net/blueva/luak/require/RequireSampleLoadLuaError.java
package net.blueva.luak.require;

import net.blueva.luak.LuaValue;
import net.blueva.luak.lib.ZeroArgFunction;

/**
 * This should fail while trying to load via 
 * "require()" because it throws a LuaError
 * 
 */
public class RequireSampleLoadLuaError extends ZeroArgFunction {
	
	public RequireSampleLoadLuaError() {		
	}
	
	public LuaValue call() {
		error("sample-load-lua-error");
		return LuaValue.valueOf("require-sample-load-lua-error");
	}	
}

File: blueluak-jvm/src/test/java/net/blueva/luak/require/RequireSampleLoadLuaError.kt
package net.blueva.luak.require

import net.blueva.luak.LuaValue
import net.blueva.luak.lib.ZeroArgFunction

/**
 * This should fail while trying to load via
 * "require()" because it throws a LuaError
 * 
 */
class RequireSampleLoadLuaError : ZeroArgFunction() {
    override fun call(): LuaValue? {
        error("sample-load-lua-error")
        return valueOf("require-sample-load-lua-error")
    }
}
